component-api/src/main/java/org/talend/sdk/component/api/record/Schema.java

```diff
@@ -45,10 +46,15 @@
     Schema getElementSchema();
 
     /**
-     * @return the entries for records.
+     * @return the data entries for records (not contains meta data entries).
      */
     List<Entry> getEntries();
 
+    /**
+     * @return All entries, including data and metadata, of this schema.
+     */
+    Stream<Entry> getAllEntries();
```

Comment:
Provide also a `getMetadata()` method that returns a `List<Entry>` as for entries.

Reply:
ok